newIDE/app/src/SceneEditor/SceneVariablesDialog.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+import VariablesEditorDialog from '../VariablesList/VariablesEditorDialog';
+
+type Props = {|
+  open: boolean,
+  layout: gdLayout,
+  onApply?: () => void,
```

Comment:
Should onApply be *not* optional?

Reply:
I made it optional to make it similar to the scene properties dialog - it too has optional `onApply` and on pressing OK, calls the function only if it exists. It does seem strange, since Esc or clicking outside the dialog does not seem to close the dialog, so the user is almost stuck if `onApply` were not defined. Should I make `onApply` mandatory in `ScenePropertiesDialog` too?

Reply:
Yes I think we can make it mandatory in `ScenePropertiesDialog` too. `onApply` was surely made optional for an historical reason, but should not be anymore :)

Reply:
Done!